enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,924 @@
+  complex network policy rules.
+* As an administrator, I want to be able to provision networks to my tenants to ensure their networks and applications
+  are natively isolated from other tenants.
+* As a user, I want to be able to request new private networks for my namespace, without having to get administrator
+  permission.
+* As a user, I want secondary networks to be able to have the same functionality as the primary network, including the
+  ability to use services, and communicate outside of the cluster (north/south).
+* As a user, I want to be able to use my own consistent IP address scheme in my network.
+
+### Goals
+
+* Provide a configurable way to indicate that a pod should be connected to a user-defined network of a specific type as a
+primary interface.
+* The primary network may be configured as a layer 3 or layer 2 type network.
+* Allow networks to have overlapping pod IP address space. This range may not overlap with the default cluster subnet
+used for allocating pod IPs on the cluster default network today.
+* The cluster default primary network defined today will remain in place as the default network pods attach to. The cluster
+default network will continue to serve as the primary network for pods in a namespace that has no user-defined network. Pods
+with user-defined networks will still attach to the cluster default network with limited access to Kubernetes system resources.
+Pods with user-defined networks will have at least two network interfaces, one connected to the cluster default network and one
+connected to the user-defined network.
+* Allow multiple namespaces per network.
+* Support cluster ingress/egress traffic for user-defined networks, including secondary networks.
+* Support for ingress/egress features where possible:
+    * EgressQoS
+    * EgressService
+    * EgressIP
+    * Load Balancer and NodePort Services, as well as services with External IPs.
+* In addition to ingress service support, there will be support for Kubernetes services in user-defined networks. The
+scope of reachability to that service as well as endpoints selected for that service will be confined to the network
+and corresponding namespace(s) where that service was created.
+* Support for pods to continue to have access to the cluster default primary network for DNS and KAPI service access.
+* Kubelet healthchecks/probes will still work on all pods.
+* OpenShift Router/Ingress will work with some limitations for user-defined networks.
+
+### Non-Goals
+
+* Allowing different service CIDRs to be used in different networks.
+* Localnet will not be supported initially for primary networks.
+* Allowing multiple primary networks per namespace.
+* Hybrid overlay support on user-defined networks.
+
+### Future-Goals
+
+* DNS lookup for pods returning records for IPs on the user-defined network. In the first phase DNS will return the pod
```

Comment:
No, your link is about services. Service DNS will not be affected by this enhancement since it remains a unique subnet range that is not a per network subnet. The text there is talking specifically pod DNS lookup.

Reply:
oh I see your point: "Accessing my-service works in the same way as other Services but with the crucial difference that redirection happens at the DNS level rather than via proxying or forwarding. " so this resolving will happen via DNS from default network and then we will get the podIP as the answer there and then proceed to connect to the pod via the user-defined network?

Reply:
Resolving the service DNS name will resolve to a cluster IP, then OVN will DNAT to the pod IP.